crates/bevy_render/src/fly_camera.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{
+    camera::{Camera, PerspectiveProjection, VisibleEntities},
+    render_graph::base,
+};
+use bevy_app::{AppBuilder, EventReader, Events, Plugin};
+use bevy_core::Time;
+use bevy_ecs::*;
+use bevy_ecs::{Bundle, Query, Res, ResMut};
+use bevy_input::{keyboard::KeyCode, mouse::MouseMotion, Input};
+use bevy_math::{Quat, Vec2, Vec3};
+use bevy_transform::components::{Rotation, Scale, Transform, Translation};
+
+/**
+Used in [CameraFlyingComponents](struct.CameraFlyingComponents.html)
+**/
+pub struct CameraFlyingOptions {

```

Comment:
Given that this is the component that enables "camera flying", I think it makes more sense to just call this `CameraFlying` (actually `FlyCamera` would be my preference)

Reply:
I see no reason to tie this system to a camera - might this simply be a `Flyable` system?  It can be attached to anything with a `Translation` and a `Rotation` to make it flyable.

Reply:
The benefit of this is if you want something like a spaceship with an attached camera - you now have the ability to fly the ship without a new system.

The danger here is that this becomes the "one and only" implementation of flying, with all of the bells and whistles thrown in.

Reply:
I think thats a good idea. It does make it slightly weirder to put it into bevy_render, but I think thats a decent spot for it, for now. We might want to eventually add a "bevy_movement" crate or something for various "common" movement types, but lets wait for the next movement implementation to pop up before doing that.

For now just rename the type and file and lets get this merged!

Reply:
Done!